### PR TITLE
FE-1273 Add RV Usage line chart

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Abstracts `LineChart` from report builder
  - Pulled out the custom tooltip that was in the LineChart file
- Adds RV usage history API call to the usage page (Blocked by AC-1809)
- Adds LineChart to RV usage section

### How To Test

- Run the app
- Visit the usage section (Until AC-1809 is resolved, this page will fail to load)

Alternative:
- Run `npm run test-integration-hibana` and run the accounts >  usage page integration tests

### To Do

- [ ] Improve testing if needed?

### CODE REVIEW

File: cypress/tests/integration/accounts/usage/usagePage.spec.js [hunk 1]
```diff
@@ -26,8 +26,14 @@ describe('The usage page', () => {
...
 
+      cy.stubRequest({
+        url: '/api/v1/usage/history',
+        fixture: 'usage/history/200.get.json',
+        requestAlias: 'usageHistoryReq',
+      });
+
       cy.visit(PAGE_URL);
-      cy.wait(['@usageReq', '@subscriptionReq', '@accountReq']);
+      cy.wait(['@usageReq', '@usageHistoryReq', '@subscriptionReq', '@accountReq']);
```
Comment (jonambas) [inline cypress/tests/integration/accounts/usage/usagePage.spec.js:36]: I wasn't sure how best to test the chart here, or if it even needed – open to ideas here

File: src/components/charts/LineChart.js [hunk 2]
```diff
@@ -12,45 +12,13 @@ import {
 import moment from 'moment';
```
Comment (jonambas) [inline src/components/charts/LineChart.js:12]: Moved `LineChart` to `src/components/charts`, but left the custom tooltip in `pages/reportBuilder`

There were no tests – let me know if I should write some

File: src/pages/usage/components/tests/RVUsageChart.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        date: '2020-08-16',
+        usage: 5678,
+      },
+    ],
+  };
+
+  it('renders a chart element correctly', () => {
+    const instance = shallow(<RVUsageChart {...defaultProps} />);
+    // Checks y axis label rendered
+    expect(instance).toHaveTextContent('Validations');
```
Comment (jonambas) [inline src/pages/usage/components/tests/RVUsageChart.test.js:22]: This is just checking the y-axis label, not sure how else to test the chart here

File: src/components/charts/LineChart.js [hunk 4]
```diff
@@ -126,7 +94,9 @@ export default function SpLineChart(props) {
     tooltipLabelFormatter = identity,
     tooltipValueFormatter = identity,
     showXAxis,
+    xAxisKey = 'ts',
```
Comment (bkemper) [inline src/components/charts/LineChart.js:97]: 👍

File: src/pages/usage/UsagePage.js [hunk 5]
```diff
@@ -6,27 +6,35 @@ import {
...
     refetch: subscriptionRefetch,
   } = useSparkPostQuery(getSubscription);
-  const isLoading =
-    accountStatus === 'loading' || usageStatus === 'loading' || subscriptionStatus === 'loading';
-  const isError =
-    accountStatus === 'error' || usageStatus === 'error' || subscriptionStatus === 'error';
+
+  // API Status Handling
+  const statuses = [accountStatus, usageStatus, usageHistoryStatus, subscriptionStatus];
+  const isLoading = statuses.includes('loading');
```
Comment (bkemper) [inline src/pages/usage/UsagePage.js:31]: 👏 Must cleaner.

File: src/pages/usage/UsagePage.js [hunk 6]
```diff
@@ -6,27 +6,35 @@ import {
...
     status: subscriptionStatus,
     refetch: subscriptionRefetch,
   } = useSparkPostQuery(getSubscription);
-  const isLoading =
-    accountStatus === 'loading' || usageStatus === 'loading' || subscriptionStatus === 'loading';
-  const isError =
-    accountStatus === 'error' || usageStatus === 'error' || subscriptionStatus === 'error';
+
+  // API Status Handling
+  const statuses = [accountStatus, usageStatus, usageHistoryStatus, subscriptionStatus];
```
Comment (bkemper) [inline src/pages/usage/UsagePage.js:30]: Can we leave `usageHistoryStatus` out of this group?  I don't think we should wait for loading and definitely not block this whole page if the history fails.

File: src/pages/usage/components/RVUsageChart.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+  return (
+    <Box borderRadius="200" padding="200" bg="gray.1000">
+      <Box fontSize="100" color="gray.200" mb="100">
+        {getTooltipLabelFormatter('day')(label)}
+      </Box>
+      <Box>
+        <Box justifyContent="space-between" alignItems="center" display="flex">
+          <Box display="inline-flex" alignItems="center">
+            <LegendCircle mr={tokens.spacing_300} color={tokens.color_blue_700} />
```
Comment (bkemper) [inline src/pages/usage/components/RVUsageChart.js:24]: Good call!

File: src/pages/usage/components/RVUsageChart.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+import { formatNumber } from 'src/helpers/units';
+import totalRecipientValidationCost from 'src/helpers/recipientValidation';
+import { getTimeTickFormatter, getTooltipLabelFormatter } from 'src/helpers/chart.js';
+import LineChart from 'src/components/charts/LineChart';
+
+export function CustomTooltip(props) {
+  const { payload, label } = props;
+
+  // There is only ever one line in this chart, so we just pick the first entry
+  const entry = payload[0] || {};
```
Comment (bkemper) [inline src/pages/usage/components/RVUsageChart.js:14]: I confirmed with a new account that we will always have at least one datapoint.  Should we populate the rest of the billing cycle with zeros?

Screen Shot 2020-12-09 at 4 07 52 PM

File: src/pages/usage/components/RVUsageChart.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+            </Text>
+          </Box>
+        </Box>
+      </Box>
+    </Box>
+  );
+}
+
+function RVUsageChart(props) {
+  const { data } = props;
```
Comment (bkemper) [inline src/pages/usage/components/RVUsageChart.js:55]: I confirmed that we will have at least one datapoint for new accounts.  Should we populate the rest of the chart for the entire billing cycle?

Screen Shot 2020-12-09 at 4 07 52 PM

**Review by bkemper — APPROVED** [review]
LGTM, I left a couple questions, but nothing that would prevent this from going out.

**Comment by bkemper** [comment]
Oh and the cypress tests are failling for Hibana build because the `useSparkPostQuery.spec.js` used the usage page to test the hook.  The fix is to add a stub for `/usage/history`.

**Pull request merged**